crates/nix_rs/src/flake/schema.rs

```diff
@@ -1,108 +1,216 @@
-//! High-level schema of a flake
-//!
-//! TODO: Consolidate with `outputs.rs`
-use std::collections::BTreeMap;
+//! Nix flake-schemas
 
+use lazy_static::lazy_static;
 use serde::{Deserialize, Serialize};
+use std::{collections::BTreeMap, fmt::Display, path::Path};
 
-use super::{
-    outputs::{FlakeOutputs, InventoryItem, Leaf},
-    System,
-};
+use crate::system_list::SystemsListFlakeRef;
+
+use super::{command::FlakeOptions, eval::nix_eval, outputs::FlakeOutputs, url::FlakeUrl};
+
+lazy_static! {
+  /// Flake URL of the default flake schemas
+  ///
+  /// We expect this environment to be set in Nix build and shell.
+  pub static ref DEFAULT_FLAKE_SCHEMAS: FlakeUrl = {
+    Into::<FlakeUrl>::into(Path::new(env!("DEFAULT_FLAKE_SCHEMAS")))
+  };
+
+  /// Flake URL of the flake that defines functions for inspecting flake outputs
+  ///
+  /// We expect this environment to be set in Nix build and shell.
+  pub static ref INSPECT_FLAKE: FlakeUrl = {
+    Into::<FlakeUrl>::into(Path::new(env!("INSPECT_FLAKE")))
+  };
+}
+
+/// Represents the schema of a given flake evaluated using [static@INSPECT_FLAKE]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct FlakeSchemas {
+    pub inventory: BTreeMap<String, InventoryItem>,
```

Comment:
Since Nix attrsets have no concept of order, shouldn't we just use `HashMap` here (and in `outputs.rs`)?

Reply:
Done in https://github.com/juspay/omnix/pull/235/commits/9de27ef22bc18001c82b358cd3bc37ed64016e59